src/org/exist/collections/MutableCollection.java

```diff
@@ -1258,6 +1258,9 @@ private void storeXMLInternal(final Txn transaction, final DBBroker broker, fina
         if(info.isCreating()) {
             info.getTriggers().afterCreateDocument(broker, transaction, document);
         } else {
+            final StreamListener listener = broker.getIndexController().getStreamListener(document, StreamListener.ReindexMode.REPLACE_DOCUMENT);
```

Comment:
@shabanovd Well I guessed you hadn't checked thoroughly otherwise I wouldn't be explaining so much.

The `listener` in the index controller is taken from the index worker, so they are effectively the same thing.